TeamCode/src/main/java/org/firstinspires/ftc/teamcode/PixyCam.java

```diff
@@ -0,0 +1,170 @@
+package org.firstinspires.ftc.teamcode;
+
+import android.util.Log;
+
+import com.qualcomm.robotcore.hardware.I2cDeviceSynch;
+
+import java.util.Arrays;
+import java.util.Collections;
+
+/**
+ * Created by NoahR on 10/3/17.
+ * Note about Modern Robotics I2C wiring. (Left to right)
+ * 0 - 1 - 2 - 3
+ * 5v - Data - Clock - gnd
+ */
+
+public class PixyCam
+{
+    I2cDeviceSynch m_deviceClient;
+    private byte[] m_registerValues;
+    private int m_dataStartAddress = 0;
+
+    private final int START_ADDRESS = 0x54;
+    private final int READ_LENGTH = 26; // Start read at 0x58, end at 0x58 + 26
+    private final byte[] PIXY_DEVICE_SIGNATURE = {85, (byte) 170, 85, (byte) 170};
+    private final int SIGNATURE_SPACE_SIZE = 10; // Device signature will be in the first 10 bytes
+
+    PixyCam() {}
+    PixyCam( I2cDeviceSynch deviceClient ) { initialize( deviceClient); }
+
+    public void initialize( I2cDeviceSynch deviceClient )
+    {
+        m_deviceClient = deviceClient;
+        m_deviceClient.engage();
```

Comment:
Set the I2cDeviceSynch ReadWindow here? Lots of optimization talk [here](https://ftctechnh.github.io/ftc_app/doc/javadoc/com/qualcomm/robotcore/hardware/I2cDeviceSynch.html)